src/components/Web3Provider/index.tsx

```diff
@@ -0,0 +1,146 @@
+import { Web3ReactProvider } from '@web3-react/core'
+import { Connector } from '@web3-react/types'
+import {
+  coinbaseWalletHooks,
+  createOrderedConnectors,
+  fortmaticHooks,
+  getConnectorForWallet,
+  gnosisSafe,
+  injectedHooks,
+  MODAL_WALLETS,
+  network,
+  Wallet,
+  walletConnectHooks,
+} from 'connectors'
+import usePrevious from 'hooks/usePrevious'
+import { useEffect, useState } from 'react'
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+import { updateWalletOverride } from 'state/walletOverride/reducer'
+
+const connect = async (connector: Connector) => {
+  try {
+    if (connector.connectEagerly) {
+      await connector.connectEagerly()
+    } else {
+      await connector.activate()
+    }
+  } catch (error) {
+    console.debug(`web3-react error: ${typeof connector}, ${error}`)
+  }
+}
+
+interface ConnectorState {
+  isActive: boolean
+  previousIsActive: boolean | undefined
+}
+
+// This component handles state changes in web3-react. It eagerly connects to all wallets.
+// It also checks for Coinbase Wallet, Wallet Connect Fortmatic or Injected wallets to become active.
+function Web3Updater() {
+  const dispatch = useAppDispatch()
+
+  const walletOverride = useAppSelector((state) => state.walletOverride.walletOverride)
+  const walletOverrideBackfilled = useAppSelector((state) => state.walletOverride.walletOverrideBackfilled)
+
+  const injectedIsActive = injectedHooks.useIsActive()
```

Comment:
Tighten up all the logic from L45-L55 and L76-113 by avoiding using as much state.

```
const injectedIsActive = injectedHooks.useIsActive()
const coinbaseWalletIsActive = coinbaseWalletHooks.useIsActive()
const walletConnectIsActive = walletConnectHooks.useIsActive()
const fortmaticIsActive = fortmaticHooks.useIsActive()
const activeMap = useMemo(() => ({
  [Wallet.INJECTED]:, injectedIsActive,
  [Wallet.COINBASE_WALLET]:, coinbaseWalletIsActive,
  [Wallet.WALLET_CONNECT]: walletConnectIsActive,
  [Wallet.FORTMATIC]: fortmaticIsActive,
}), [injectedIsActive, coinbaseWalletIsActive, walletConnectIsActive, fortmaticIsActive])
const previousActiveMap = usePrevious(activeMap)

...

useEffect(() => {
  Object.entries(activeMap).forEach(([wallet, active]) => {
    if (active && !previousActiveMap[wallet]) {
      if (isEagerlyConnecting) {
        setIsEagerlyConnecting(false)
      } else if (!walletOverrideBackfilled) {
        // When a user manually sets their new connection, set a wallet override.
        // Also set an override when they were a user prior to this state being introduced.
        dispatch(updateWalletOverride({ wallet }))
      }
    }
  })
}, [...])
```